extensions/ql-vscode/src/remote-queries/run-remote-query.ts

```diff
@@ -451,3 +483,20 @@ async function buildRemoteQueryEntity(
     repositoryCount,
   };
 }
+
+function getQueryName(queryMetadata: QueryMetadata | undefined, queryFilePath: string): string {
+  // The query name is either the name as specified in the query metadata, or the file name.
+  return queryMetadata?.name ?? path.basename(queryFilePath);
+}
+
+async function getControllerRepoId(credentials: Credentials, owner: string, repo: string): Promise<number> {
+  try {
+    return await ghApiClient.getRepositoryIdFromNwo(credentials, owner, repo);
```

Comment:
Would it make sense to drop the `owner` and `repo` parameters from this method and instead call `getRemoteControllerRepo` in this method? The only thing that now differentiates this from the `getRepositoryIdFromNwo` method is the way error messages are handled; it doesn't actually just get the controller repo ID, you can pass any arbitrary owner/repo combination into it.

Reply:
I think that would be a good way of cleaning this code up by putting more implementation into a method. Unfortunately it's not quite as clean as it could be, mostly because we use the `owner` and `repo` fields later on in `runRemoteQuery` so we have to pass those back as well. But a nice way of doing that could be changing to passing around a `Repository` object instead of just the ID or owner or repo name separately. I've had a go at this and I think it's worked out quite nicely. It's slightly more code changes than strictly necessary but I think it's all going in the right direction.